pkg/report/report.go

```diff
@@ -199,13 +207,13 @@ func (r *report) processResults() {
 	}
 	r.total = time.Since(st)
 
-	r.rps = float64(len(r.lats)) / r.total.Seconds()
-	r.average = r.avgTotal / float64(len(r.lats))
+	r.average = r.avgTotal / r.requests
```

Comment:
separate weighted average, `r.weightedAverage = ...`

Also isn't this applying the weights twice? avgTotal is already divided by weight in `r.avgTotal += dur.Seconds / weight`?